### PR TITLE
fix(text): handle code points > U+FFFF in `levenshteinDistance`

### PR DESCRIPTION
**Pull request by lionel-rowe**
Fixes https://github.com/denoland/std/issues/6013

### CODE REVIEW

**Comment by lionel-rowe** [comment]
Benchmarks:

```ts

import { levenshteinDistance as current } from "jsr:@std/text@1.0.6/levenshtein-distance";
import { levenshteinDistance as next } from "https://raw.githubusercontent.com/lionel-rowe/std/levenshtein-unicode/text/levenshtein_distance.ts";

for (const [name, fn] of Object.entries({ current, next })) {
  Deno.bench(`${name} (ASCII)`, () => {
    fn("a".repeat(10), "b".repeat(10));
    fn("a".repeat(100), "b".repeat(100));

    fn("a".repeat(10), "");
    fn("a".repeat(100), "");
    fn("", "b".repeat(10));
    fn("", "b".repeat(100));

    fn(
      "a".repeat(100) + "b".repeat(100) + "a".repeat(100),
      "b".repeat(100) + "a".repeat(100) + "b".repeat(100),
    );
  });
}

for (const [name, fn] of Object.entries({ current, next })) {
  // will give wrong result with `current`, but just testing perf here
  Deno.bench(`${name} (with emoji)`, () => {
    fn(
      "a".repeat(100) + "💩".repeat(100) + "a".repeat(100),
      "💩".repeat(100) + "a".repeat(100) + "💩".repeat(100),
    );
  });
}
```

Performance is almost identical on my machine, typical run:

image

File: text/levenshtein_distance.ts [hunk 1]
```diff
@@ -4,20 +4,23 @@ const { ceil } = Math;
 
 // This implements Myers' bit-vector algorithm as described here:
 // https://dl.acm.org/doi/pdf/10.1145/316542.316550
-const peq = new Uint32Array(0x10000);
+const peq = new Uint32Array(0x10ffff);
```
Comment (0f-0b) [inline text/levenshtein_distance.ts:7]:
```suggestion
const peq = new Uint32Array(0x110000);
```

Maybe add a test for `levenshteinDistance("\u{10FFFF}", "\u{10FFFF}")`?

**Comment by lionel-rowe** [comment]
Previous fix was still buggy as it's tough to keep the various increments and length measurements in sync, given that sometimes you want the code point length and sometimes the code unit length. As a result I just switched to using `[...str]` char arrays, which surprisingly still has no measurable impact on perf (updated benchmarks above).

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**